elegant/process_experiment.py

```diff
@@ -80,11 +81,23 @@ def segment_experiment(experiment_root, model, channels='bf', use_gpu=True, over
         overwrite_existing: if False, the segmenter will not be run on existing
             mask files, nor will existing annotations be modified even if new
             mask files are generated for a timepoint.
+        timepoint_filter - filter for scan_experiment_dir specifying timepoints to segment
+        mask_root - root directory for masks that will be used to updating annotations
     """
+    if mask_root is None:
+        mask_root = experiment_root / 'derived_data' / 'mask'
+
     experiment_root = pathlib.Path(experiment_root)
-    positions = load_data.scan_experiment_dir(experiment_root, channels=channels)
-    mask_root = experiment_root / 'derived_data' / 'mask'
-    segment_images.segment_positions(positions, model, mask_root, use_gpu, overwrite_existing)
+    positions = load_data.scan_experiment_dir(experiment_root, channels=channels, timepoint_filter=timepoint_filter)
+    if positions:
+        process = segment_images.segment_positions(positions, model, mask_root, use_gpu, overwrite_existing)
+        if process.stderr:
```

Comment:
You probably want to rethink your overall logging and error-reporting scheme. Printing warnings is pretty useless in the case where the segmenter gets run automatically by the timepoint code, and in other non-interactive uses. If it's useful to have a logfile of when segmentation succeeds, you should also add to the same logfile the failure cases.

Reply:
The use case here is not to print warnings, but to print errors. As it is now, errors (via stderr) are stashed away from the user. This lack of notification creates the scenario where the user goes forward with further processing down the pipeline without catching a potential fail at this step; later, the user has to track backwards to find that the pipeline simply did not work the way they think it did at this early step of preprocessing. The hope is to make all erroring by the segmenter transparent so that the user is aware of fails before proceeding further.

Reply:
Logic still applies to printed "errors" vs. "warnings". How is stdout less "stashed away" than "stderr" in non-interactive cases? Or is this for interactive use? Let's talk in person about how logging is handled, and what cases need to be addressed better. Right now, when run by the timepoint code, stdout and stderr are both redirected to the "segmentation.log" file, which is opened in append mode.